_pages/extension_sharing_strategies.md

```diff
@@ -0,0 +1,143 @@
+
+
+## 3. Extension process (create, deploy, update)
+
+The figure below illustrates the process for (1) creating a new extensions, (2) creating a recipe to register an extension with the extension catalog, and (3) to update an extension/recipe. The figure also illustrates the automated CI processes that are managed in the [nwb-extensions](https://github.com/nwb-extensions) catalog. The catalog process is modeled after the conda-forge model, enabling us to largely automate the catalog process using free, public services and avoiding the need to host our own services.
+
+<img alt="NDX Catalog Logo" src="images/nwbn_extension_catalog_workflow.png" width="800" class="center-block">
+
+### 3.1 Continuous Integration (CI) for Extensions
+
+This can be separated in CI for three main repositories:
+
+1. the staged-extensions repository as part of the nwb-extensions catalog,
+1. the ndx-custom-recipe repo as part of the nwb-extensions catalog,  and
+1. the users ndx-custom repository.
+
+Part 1 and 2 are part of the nwb-extensions catalog and are maintained by the NWB:N development team. Part 3 will need to be setup and maintained by the extension developers. The ndx-template can provide templates for defining CI for ndx-custom repositories, but ultimately this will be up to the extension developers to do.
+
+#### 3.1.1 staged-extensions CI
+
+For the staged-extensions repositories we have common tests to validate basic compliance and working order of the extensions before creating the ndx-custom-recipe repo. Here we can test, e.g., the extensions metadata, i.e., are all required fields for the catalog specified and valid.
+
+#### 3.1.2 ndx-custom-recipe CI
+
+Similar to the staged-extensions, we can provide CI for ndx-custom-recipe repositories to automatically test compliance and working order of the extensions before merging pull requests. Here we can test, e.g., the extensions metadata, i.e., are all required fields for the catalog specified and valid.
+
+#### 3.1.3 ndx-custom CI
+
+Here we can provide example configuration files and instructions as part of the ndx-template repository but ultimately this is up to the user to setup and maintain. The ndx-template repo itself will likely not include CI itself. However, it includes templates for setting up CI for ndx-custom user extension repositories. See the [ndx-template](https://github.com/nwb-extensions/ndx-template) repo for details. We generally recommend to use CI to:
```

Comment:
We are providing the azure yaml in the template. What else is there?

Reply:
Maybe we can also provide an azure yaml within `{{ cookiecutter.namespace }}/`

Reply:
Not sure right now. That's a question for @rly

Reply:
I suggest we provide templates and setup instructions for the user to run CI on their generated extension repo.